File: jsimple-json/src/main/java/jsimple/json/readerwriter/JsonObjectProperty.java
package jsimple.json.readerwriter;

/**
 * @author Bret Johnson
 * @since 5/22/13 9:12 PM
 */
public class JsonObjectProperty extends JsonProperty {
    public JsonObjectProperty(String name, int id) {
        super(name, id);
    }

    public JsonObjectReader readValue(JsonObjectReader objectReader) {
        return (JsonObjectReader) objectReader.readPropertyValue();
    }

    public JsonObjectWriter write(JsonObjectWriter objectWriter) {
        return objectWriter.writeObjectProperty(this);
    }
}

File: jsimple-json/c#/jsimple/json/readerwriter/JsonObjectProperty.cs
namespace jsimple.json.readerwriter
{

	/// <summary>
	/// @author Bret Johnson
	/// @since 5/22/13 9:12 PM
	/// </summary>
	public class JsonObjectProperty : JsonProperty
	{
		public JsonObjectProperty(string name, int id) : base(name, id)
		{
		}

		public virtual JsonObjectReader readValue(JsonObjectReader objectReader)
		{
			return (JsonObjectReader) objectReader.readPropertyValue();
		}

		public virtual JsonObjectWriter write(JsonObjectWriter objectWriter)
		{
			return objectWriter.writeObjectProperty(this);
		}
	}

}